plugins/needs_restarting.py

```diff
@@ -126,6 +127,28 @@ def print_cmd(pid):
     print('%d : %s' % (pid, command))
 
 
+def get_service_dbus(pid):
+    bus = dbus.SystemBus()
+    systemd_manager_object = bus.get_object(
+        'org.freedesktop.systemd1',
+        '/org/freedesktop/systemd1'
+    )
+    systemd_manager_interface = dbus.Interface(
+        systemd_manager_object,
+        'org.freedesktop.systemd1.Manager'
+    )
+    service_proxy = bus.get_object(
+        'org.freedesktop.systemd1',
+        systemd_manager_interface.GetUnitByPID(pid)
+    )
+    service_properties = dbus.Interface(
+        service_proxy, dbus_interface="org.freedesktop.DBus.Properties")
+    name = service_properties.Get(
+        "org.freedesktop.systemd1.Unit",
+        'Id'
+    )
+    return name
```

Comment:
I believe this will now return all unit types not just services for given pid (in my local test it also printed `session-2.scope`).

We probably want to check if the suffix of the `name` is ".service", I think this should be safe because the man page for `systemd.unit` specifies: 
> The type suffix must be one of ".service", ".socket", ".device", ".mount", ".automount", ".swap", ".target", ".path", ".timer", ".slice", or ".scope".